lib/src/link/types.ts

```diff
@@ -0,0 +1,58 @@
+type Space = "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | "xxlarge";
+type Margin = {
+  top?: Space;
+  bottom?: Space;
+  left?: Space;
+  right?: Space;
+};
+
+type Props = {
+  /**
+   * If true, the color is inherited from parent.
+   */
+  inheritColor?: boolean;
+  /**
+   * If true, the link is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * @Deprecated Source of the icon.
+   */
+  iconSrc?: string;
+  /**
+   * Element used as the icon that will be placed next to the link text.
+   */
+  icon?: any;
+  /**
+   * Indicates the position of the icon in the component.
+   */
+  iconPosition?: "before" | "after";
+  /**
+   * Page to be opened when the user clicks on the link.
+   */
+  href?: string;
+  /**
+   * If true, the page is opened in a new browser tab.
+   */
+  newWindow?: boolean;
+  /**
+   * If defined, the link will be displayed as a button. This 
+   * function will be called when the user clicks the link.
+   */
+  onClick?: () => void;
+  /**
+   * Link text.
+   */
+  text?: string;
```

Comment:
I consider `text` and `icon` both necessary props. I mean, Link component must have one of them, or both, but never any of them. For further understanding: 

![image](https://user-images.githubusercontent.com/44321109/152501068-641c8b19-d5f7-4b6a-b062-546906861a21.png)

Reply:
Is really icon a required prop? I see that text is needed but I don't see it very clear with the icon.

Reply:
Neither is strictly required. What it is required is to have a Link with at least one of this two props (`text` or `icon`, or both, never none of them). This can be achieved with type Overload. Here is an [example](https://www.typescriptlang.org/es/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wFgAoCgeirgBUBPMJAZwpiaTgBlgA7ANZ0kADxgAFHGBZwAvHADeFOHBiiYAfgBccFjCj8A5gG5lcYGgh8deg3xMUAvqfIdmPfgICSlvpIjScojEMAB0AIJ8aAAW0AASdACy3OEw+sAARgCuaiwAPAnJkTHQAKIANkggSHwwAHxwAGSKZmpiNun2LioWVtq6nQ7kzuyccADyAG5IUOUQKAAmQUrkKgAUYFIsOryCwmL+0gCUOgBSAMoAGqEVVTUw3XAbWzuePlaHLCdwF9e31bUXCNKOQaHAAMK4SB8e4UXx6DyCHRTGZzRZBZ4BbaIoTqT5wAA+OPefi231+N0qAPgsgaKxURBgWSgfDgeWyaSsdQUmyxoTaMEcPK2-PUjjyVA5MC5QJc1FoAFUWChDEgKJgslEYMArHBwmAwGsji1VnBGczWWszCo8nVrSpFFQAFRwQC8G4BaPbgpSgOCgcCdVEc9ptuwEcCodtNDsdLtdgDg9vXlcpwQwQCBLANBqMOvKh1TqWQAIhQhfDkejKgUzrdCfCSZTaYzgeDbLzvT4RfbpYjLarsdr9dT6f9zezIc8+bERZL5l8nd83fLOZ7pqOQKAA) I've done.

Reply:
Changed types to adjust to theses cases. @GomezIvann please check them.